File: CS/TreeListCellMerging/MyPrinter.cs
using DevExpress.XtraTreeList;
using DevExpress.XtraTreeList.Printing;

namespace TreeListCellMerging {
    public class MyPrinter : TreeListPrinter {

        public MyPrinter(TreeList treeList)
            : base(treeList) {

        }


        protected override DevExpress.XtraTreeList.Nodes.Operations.TreeListOperationPrintEachNode CreatePrintEachNodeOperation() {

            if (TreeList.OptionsPrint.PrintAllNodes)
                return new MyTreeListOperationPrintEachNode(TreeList, this, TreeList.ViewInfo, TreeList.OptionsPrint.PrintTree, TreeList.OptionsPrint.PrintImages);
            return new MyTreeListOperationPrintEachNode(TreeList, this, TreeList.ViewInfo, TreeList.OptionsPrint.PrintTree, TreeList.OptionsPrint.PrintImages);
        }
    }
}

File: VB/TreeListCellMerging/MyPrinter.vb
Imports DevExpress.XtraTreeList
Imports DevExpress.XtraTreeList.Nodes.Operations
Imports DevExpress.XtraTreeList.Printing

Namespace TreeListCellMerging
	Public Class MyPrinter
		Inherits TreeListPrinter

		Public Sub New(ByVal treeList As TreeList)
			MyBase.New(treeList)

		End Sub


		Protected Overrides Function CreatePrintEachNodeOperation() As DevExpress.XtraTreeList.Nodes.Operations.TreeListOperationPrintEachNode
			Return New MyTreeListOperationPrintEachNode(TreeList, Me, TreeList.ViewInfo, TreeList.OptionsPrint.PrintTree, TreeList.OptionsPrint.PrintImages, TreeList.OptionsPrint.PrintCheckBoxes, TreeList.OptionsPrint.PrintAllNodes)

		End Function
	End Class
End Namespace
